pkglistgen.py

```diff
@@ -1123,7 +1123,7 @@ def do_solve(self, subcmd, opts):
 
     @cmdln.option('-f', '--force', action='store_true', help='continue even if build is in progress')
     @cmdln.option('-p', '--project', help='target project')
-    @cmdln.option('-s', '--scope', default='all', help='scope on which to operate ({}, staging:$letter)'.format(', '.join(SCOPES)))
+    @cmdln.option('-s', '--scope', action='append', help='scope on which to operate ({}, staging:$letter)'.format(', '.join(SCOPES)))
```

Comment:
do makes sense to maintain the 'all' as `default`? Maybe needs to be changed to `[all]` or something

Reply:
like this?

```
diff --git a/pkglistgen.py b/pkglistgen.py
index f72e24f..4f0cbf0 100755
--- a/pkglistgen.py
+++ b/pkglistgen.py
@@ -1123,7 +1123,7 @@ class CommandLineInterface(ToolBase.CommandLineInterface):
 
     @cmdln.option('-f', '--force', action='store_true', help='continue even if build is in progress')
     @cmdln.option('-p', '--project', help='target project')
-    @cmdln.option('-s', '--scope', action='append', help='scope on which to operate ({}, staging:$letter)'.format(', '.join(SCOPES)))
+    @cmdln.option('-s', '--scope', action='append', default=['all'], help='scope on which to operate ({}, staging:$letter)'.format(', '.join(SCOPES)))
     @cmdln.option('--no-checkout', action='store_true', help='reuse checkout in cache')
     @cmdln.option('--stop-after-solve', action='store_true', help='only create group files')
     def do_update_and_solve(self, subcmd, opts):
@@ -1140,7 +1140,7 @@ class CommandLineInterface(ToolBase.CommandLineInterface):
         opts.staging_project = None
 
         # special case for all
-        if len(opts.scope) == 0 or opts.scope == ['all']:
+        if opts.scope == ['all']:
             opts.scope = self.SCOPES[1:]
 
         for scope in opts.scope:

Reply:
Cool, yes